buildSrc/src/test/resources/input/additional_props.json

```diff
@@ -20,8 +20,14 @@
     },
     "flags": {
       "additionalProperties": {
+        "type": "boolean"
+      }
+    },
+    "tags": {
+      "additionalProperties":  {
         "type": "string"
```

Comment:
the way I saw in the `extractAdditionalProperties` function you only support `Boolean` and `Map<*,*>`. Am I missing something ?

Reply:
Because the type of additionalProperties in the `JsonDefinition` data class is set to `Any`, when `additionalProperties` is an object, it'll be converted to a generic map, instead of a typed Kotlin object. 
`"additionalProperties": true` will translate to a boolean, and `"additionalProperties": { "type": "string" }` will translate to a map. 
Then in the `extractAdditionalProperty` method, when I see it's a map, I look at the `"type"` key of the map to know what's the expected type for additional properties